dask/array/core.py

```diff
@@ -590,7 +593,13 @@ def map_blocks(
     original_kwargs = kwargs
 
     if dtype is None and meta is None:
-        dtype = apply_infer_dtype(func, args, original_kwargs, "map_blocks")
+        try:
+            meta = func(
+                *[getattr(arg, "_meta", arg) for arg in args],
+                **{k: getattr(v, "_meta", v) for k, v in original_kwargs.items()}
+            )
```

Comment:
@pentschev it would be good to get your thoughts on this.  

This seems to work fine, but it seems like something that you would have thought of before, so I suspect that there is some reason why it was not done.

Reply:
It seems very similar to what `compute_meta` is doing, specifically https://github.com/dask/dask/blob/master/dask/array/utils.py#L113-L116. Maybe you wanted to use `compute_meta` instead, or is there something different here that I'm missing?

If the question instead is "why was there no support for `map_blocks` yet?", then probably it's part of the incomplete work for `_meta` support, as tracked in https://github.com/dask/dask/issues/4888.